QuestionAnswering/QA_ICL/README.md

```diff
@@ -1,15 +1,199 @@
+  - `fastapi_app`: A FastAPI application.
+  - `triton_inference_server`: Serves the Sentence-BERT model for generating text embedding.
+- `frontend/`: The frontend, called Marie Frontend, serves as the user interface for interacting with the backend services. Built using Next.js, it communicates with the backend to perform operations such as querying chemical species and filtering zeolites, and displays the results to users in a user-friendly manner.
+
+## Getting Started
+
+### Prerequisities
+ - Docker (recommended for deployment)
+ - Python 3.8 or higher
+ - Node.js and npm (for frontend development)
+
+### Deployment Setup
+1. Clone the TWA repository and go to the QA_ICL folder
+
+    ```
+    cd TheWorldAvatar/QuestionAnswering/QA_ICL/
+    ```
+2. Backend Setup
+   - Follow steps [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend#docker-deployment-recommended), which covers the following topics
+      - Populate data for `Fastapi App` and `triton`
+      - Ingest data to `Fastapi App` and `triton`
+      - Launch `Fastapi App` and `triton` containers
+   - Configuration
+     - Fastapp Credentials: create file `app.local.yaml` as instructed [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend/fastapi_app#configurable-parameters) and provide credentials
+     - Port mapping: [docker-compose.yaml](./backend/docker-compose.yaml)
+3. Frontend Setup
+   - Follow steps [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/frontend/next_app_marie#deployment-via-docker) to deploy `NextApp` as docker container
+     - Build image
+     - Launch `NextApp` container
+   - Configuration
+     - `BASE_PATH`: [.env.production](./frontend/next_app_marie/.env.production)
+     - Backend endpoint: [.env](./frontend/next_app_marie/.env)
+4. Nginx Redirection Setup
+   - Create redirection for the backend endpoint. Trailing forward slash `/` is allowed.
+      ```
+      location /BACKEND_PATH/ {
+        proxy_pass          http://HOST:PORT/PATH/;
+        ...
+      }
+      ```
+   - Create redirection for the frontend endpoint. Trailing forward slash `/` is **NOT allowed**.
+      ```
+      # correct
+      location /FRONTEND_PATH {
+        proxy_pass          http://HOST:PORT/PATH;
+        ...
+      }
+
+      # wrong
+      location /FRONTEND_PATH/ {
+        proxy_pass          http://HOST:PORT/PATH/;
+        ...
+      }
+      ```
+   - Update backend entpoint configuration in NextApp `.env` file to the redirection address
+   - Rebuild Image for `NextApp` and launch the container
+   - Troubleshoot
+     - CORS error when submitting question: 
+       - This is caused by incorrect setting of the backend redirection
+
+
+### Development Setup
+1. Clone the TWA repository and go to the QA_ICL folder
+
+    ```
+    cd TheWorldAvatar/QuestionAnswering/QA_ICL/
+    ```
```

Comment:
My opinion is that the development setup should not be in this README, because the development of each component has a different set of dependencies:
- data utility scripts: no dependencies on other components
- backend: technically no dependencies, but might make use of data derived from the data utility scripts, which have been explained the backend README.
- frontend: dependencies on the backend; it should be mentioned that the dev frontend can connect to either a backend in deployment or development, or the mock server.

Someone working solely on the frontend can just deploy a local backend and work with that without having to install the backend in the host machine.